src/main/webapp/script.js

```diff
@@ -249,3 +257,27 @@ function createRating(rating) {
   
   return ratingDiv;
 }
+
+window.onbeforeunload = function() {
+  localStorage.setItem("listings", JSON.stringify(listingsLocalStorage));
+  localStorage.setItem("location", locationQuery);
+  localStorage.setItem("zipcode", document.getElementById('zipCode').value);
+  localStorage.setItem("product", document.getElementById('product').value);
+}
+
+window.onload = function() {
+  listingsLocalStorage = JSON.parse(localStorage.getItem("listings"));
+  locationQuery = localStorage.getItem("location");
+  bounds = new google.maps.LatLngBounds();
+  let zipcode = localStorage.getItem("zipcode");
+  let product = localStorage.getItem("product");
+
+  if(listingsLocalStorage != null && locationQuery != null && zipcode != null){
```

Comment:
nit: Please add spaces here. It should be 
```
 if (listingsLocalStorage != null && locationQuery != null && zipcode != null) {
```

Reply:
fixed